From 865e9711509b70ded8b25751ab0f2f9c054f8c5b Mon Sep 17 00:00:00 2001
From: Matt Ranostay <matt.ranostay@intel.com>
Date: Tue, 9 Feb 2016 10:58:18 -0800
Subject: [PATCH 3/5] usb: xhci-intel-cap: prevent NULL dereference

Add NULL deference checks in xhci_intel_phy_mux_switch()..

Change-Id: I32bba8c243eb940a95494808c00bf7f8ab3c3b4b
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-18399
Signed-off-by: Matt Ranostay <matt.ranostay@intel.com>
Reviewed-on: https://android.intel.com:443/468227
---
 drivers/usb/host/xhci-intel-cap.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/host/xhci-intel-cap.c b/drivers/usb/host/xhci-intel-cap.c
index d30d982..608bd7f 100644
--- a/drivers/usb/host/xhci-intel-cap.c
+++ b/drivers/usb/host/xhci-intel-cap.c
@@ -163,8 +163,10 @@ int xhci_intel_phy_mux_switch(struct xhci_hcd *xhci, int is_device_on)
 	unsigned long	timeout;
 	u32		data;
 
-	if (!xhci || !xhci->phy_mux_regs)
+	if (!xhci || !xhci->phy_mux_regs) {
 		pr_err("No XHCI or Not support phy mux capability\n");
+		return -EINVAL;
+	}
 
 	xhci_dbg(xhci, "XHCI phy mux switch to %s path\n",
 				is_device_on ? "dev" : "Host");
-- 
1.9.1

